### DISCUSSION TITLE
Asistencia Técnica NVIM

### DISCUSSION THREAD

**Discussion by feremmed** [adalessa/dotfiles#3] [Q&A] [answered]
Hola genio!
Te cuento que te empecé a seguir en YT y me copa tu canal, sos un bocho, gracias por subir todo ese contenido, me ayuda mucho. 
Te escribo para pedirte algo de ayuda ya que en uno de tus videos mencionabas que abrías este canal para consultas.

Antes de explicarte mi problema te doy un poco de contexto:
Soy un tipo de 34 años de oficio Electricista, que se le dio por aprender programación, lo bueno es que voy bien y me gusta, lo malo es que en el proceso me mando algunas cagadas que quiero creer son propias de mi nivel bajo, je!.  El otro gran inconveniente es que tengo una compu muy limitada por ahora, pero le ponemos garra y bancamos la parada (ella y yo) siempre pa' lante y sin miedo al éxito XD !!   Tan limitada es la pobrecita que entedí (según mi criterio) que mi mejor ruta de aprendizaje sería primero VIM/NEOVIM porque cualquier otro editor me la haría explotar. Con el plus de que Nvim mejoró mi dinamismo en el teclado, lo usé bastante (mientras andaba) y me resultó genial casi ni tocar el mouse.  Cabe aclarar que comencé a aprender desde un nivel muy básico, y Nvim aceleró mi proceso de aprendizaje, o sea que obtuve mas beneficios que solo un editor ultra liviano y lo quiero seguir usando.

El Problema:
Viendo uno de tus videos de configuración de Nvim, se me ocurrió la gran idea de copiar la tuya, pero por salame rompí todo, ahora entiendo que init.lua es mas peligroso y complejo para mi que el viejo y confiable init.vim
Sospecho que es por haber metido LUA en Windows terminal, o quizás nada tenga que ver, pero todas las listas de errores que saltan hacen referencia a lua,.
 Sé que es de panchos usar windows, pero por favor no olvides lo que mencione antes; estoy aprendiendo y aún no me da la sangre para instalar Linux porque de seguro haré explotar mi casa o toda la cuadra. 
Te paso unas capturas de los errores y luego te rogaré que me ayudes por favor, entiendo que te pueda llegar a dar asco windows y no tengas ganas ni de mirar, pero si no lo arreglo no tengo como seguir avanzando en mis estudios y la pc que tengo por ahora seguirá siendo la única.
Intenté solucionarlo con mis pocas herramientas de conocimientos:
Desintalé todo; vim, neovom, windows terminal, y volvi a instalar 
Reinicié
Cambie a otras configuraciones anteriores sin lua (en la captura que te mando veras otra de init.vim)
borre todo lo que diga lua 
...y nada
me saltan todos esos errores y además no se me colorean las palabras (syntax creo que es)
ojalá sea una boludez de solucionar, pero solo no puedo.

Por favor invoco tus poderes, sabiduría, piedad y gran misericordia OH SALVE GRAN ALPHA DEVELOPER (*Japish! de tirarse al piso a implorar)

![CagadaFer](https://user-images.githubusercontent.com/93925739/141509182-722741cf-7fcb-45a7-9782-4ec571078bdb.png)

**Comment by adalessa** [comment]
Hola @feremmed gracias por todos lo que decis me gusta poder ayudar a la gente mas a la que tiene ganas de aprender.
No tiene nada de malo usar windows ni nada. asi que vamos por pasos para tratar de diagnosticar un poco que es lo que esta pasando.
Primero que nada que version estas usando exactamente de neovim.
Es necesario que tengas la version 0.5.0 o superior. 
Tambien comentame como lo instalaste, para ver si eso ayuda a investigar un poco.

Buscando el error encontre esto 
https://issueexplorer.com/issue/vim-airline/vim-airline/2334
el problema de esa persona tenia que ver en la instalacion, si bien era en ubuntu ver si en tu caso es igual o que puede ser.

Espero tu respuesta para tratar de ayudar te a solucionar tu problema.

**Comment by feremmed** [comment]
Hola.
    Grata sorpresa me llevé al ver tu mensaje, pensé que era tan NOOB que
no me darías ni bola XD!!

    Por hacerme el canchero me baje la version 6 -pre, ahora no se si fue
buena decisión y por obvias razones windowsisticas debí bajarmelo con el
Chocolatey

    En el enlace que me mandaste veo que el problema que allí muestran es
idéntico al mío, pero no veo que le planteen una solución alguna al fulano
con mi misma desgracia, o yo no la estaría encontrando.

    La otra cosa que me cuesta entender es por que hace referencia a que
falta o no accede a una archivo 'syntax.vim' de una carpeta fuente de VIM,
cuando se supone que estoy en NEOVIM.

     Espero resolverlo pronto y no sacarte mucho tiempo.

Más allá que mi asunto se arregle o no, ya te estoy muy agradecido por la
humildad, la buena voluntad y tomarte el tiempo de responderme.



                 MUCHISIMAS GRACIAS!!

**Reply by adalessa** [reply]
Hola, dale arme esto para ayudar a la gente que quiera aprender.
Buena ya teniendo la version 6 aclara pero si no tenemos la resolucion clara pero cosas para empezar.

Una posible causa que pienso a que esta llamando a carpetas q no sabe puede ser q un plugin este buscando dependencias no completadas de librerias externas, como python.

un comando muy util de neovim es `:checkhealth` que te muestra si hay algun problema con neovim.
tambien para descargar problemas de plugins puedes iniciar neovim con el flag `--noplugin` que no va a cargar ningun plugin.

Tambien si podes subi tu config a github y asi si siguen habiendo problemas puedo tratar de replicarlo y vemos de solucionarlo.

Saludos

**Comment by feremmed** [comment, marked as answer]
HOLA!!

     Según me dijo el "checkhealth" muchos quilombos están asociados a
plugins y se me ocurrió borrarlos a todos "PlugClean" y sorprendentemente
(para mi) la mayoría de los errores de la lista roja se fueron, ahora me
doy cuenta que tenía instalados muchos plugins que no solo no necesito, si
no que ni siquiera sabía para que servian pero en la manija de usar
vim/neovim y la ignorancia se me fue la mano poniendo plugins por ver a
otros y la curiosidad de que hacen.

     Ahora entiendo que muchos plugins son incompatibles entre si y entre
vim y Neovim, también veo que me parece tan estupido el problema que no lo
puedo creer, pero en estos días estuve viendo videos de programadores y la
mayoría coinciden en que a pesar de su experiencia todos se equivocan mucho
y a diario. De todas formas no me flagelo por el error por que lo entiendo
como algo natural y también pasa en mi oficio. Pero me toca ser novato aquí
y volver a experimentarlo.

     Lo único que sigue fallando es que me pide ese archivo "syntax.vim"
que es originario de VIM, así que lo que hice fue crear el archivo que me
pide (teniendo instalado vim) que encima las direcciones de carpetas no
coinciden con las que vim viene armado, así que las armé, en fin, la
cuestión es que ahora funciona, no se si como debiera reglamentariamente
pero anda y te lo agradezco infinitamente, porque esta boludez me comió
mucho tiempo y me ayudaste a continuar.

    Te cuento que soy un porteño que huyó del ruido de la ciudad hacia el
sur de  la provincia de Salta, (Metán) y acá encontré tranquilidad, la
posibilidad de desarrollarme yo y mi familia, soy padre de dos pequeños
varoncitos y compañero de una mujer muy amorosa y buena madre. Si alguna
vez andas por acá, me encantaría recibirte y conozcas este hermoso lugar, o
aunque sea si andas de pasada hacia otro lado, pases por casa a saludar y
te agasajamos con algo rico.
Acá tenés un amigo y un tipo que te admira.

Muchisimas muchisimas gracias por tu buena voluntad.

**Comment by adalessa** [comment]
@feremmed me alegro mucho que hayas podido solucionar el problema, lo de pelearse con problema es el cotidiano para un desarrollador, no siempre se sabe lo que se hace, muchas veces es prueba y error e ir pasando. En los videos no pasa porque trato que sean dinamicos, pero incluso grabando me pasa de cruzarme con errores que me llevan mucho tiempo de resolver y terminan aveces siendo cosas simples.

Lo importante es darle para adelante y seguir. Meter conocimientos siempre es bueno no solo mantiene activo a la mente y ademas programacion es una habilidad muy solicitada actualmente.

Con respecto a lo de portenio, fui portenio, naci en zona oeste, de ahi me mude a caballito, pero hace un anio me fui del pais, hoy vivo en Alemania, no prometo nada pero si algun dia que vaya de visita y se puede un brindis se puede hacer.

Y nos estamos viendo por el canal, o por aca. siempre podes iniciar una discucion aca, en nota en eso mencioname con el @ asi me llega notificacion, sino me pasa como esta que me tomo un par de dias hasta que entre y revise porq nunca me aviso github.

Abrazo y muchos exitos
